utilities/rpp-unittests/HOST/Single_host.cpp

```diff
@@ -0,0 +1,515 @@
+                                      coordinates[2], coordinates[3], handle);
+    case 21:
+        test_case_name = "Hue modification";
+        rppi_hueRGB_u8_pkd3_host(input, srcSize[0], output, hueShift, handle);
+        break;
+    case 22:
+        test_case_name = "Saturation";
+        rppi_saturationRGB_u8_pkd3_host(input, srcSize[0], output, saturationFactor, handle);
+        break;
+    case 23:
+        test_case_name = "Histogram Balance";
+        rppi_histogram_balance_u8_pkd3_host(input, srcSize[0], output, handle);
+        break;
+    case 24:
+        test_case_name = "RandomShadow";
+        rppi_random_shadow_u8_pkd3_host(input, srcSize[0], output, coordinates[0], coordinates[1],
+                                        coordinates[2], coordinates[3], numbeoOfShadows, maxSizeX, maxSizey, handle);
+        break;
+    case 25:
+        test_case_name = "RandomCropLetterBox";
+        rppi_random_crop_letterbox_u8_pkd3_host(input, srcSize[0], output, dstSize[0], coordinates[0], coordinates[1],
+                                                coordinates[2], coordinates[3], handle);
+
+        break;
+    case 26:
+        test_case_name = "Absolute Difference";
+        rppi_absolute_difference_u8_pkd3_host(input, input_second, srcSize[0], output, handle);
+        break;
+    case 27:
+        test_case_name = "Accumulate";
+        rppi_accumulate_u8_pkd3_host(input, input_second, srcSize[0], handle);
+        break;
+    case 28:
+        test_case_name = "Accumulate Squared";
+        rppi_accumulate_squared_u8_pkd3_host(input, srcSize[0], handle);
+        break;
+    case 29:
+        test_case_name = "Accumulate Weighted";
+        rppi_accumulate_weighted_u8_pkd3_host(input, input_second, srcSize[0], alpha, handle);
+        break;
+    case 30:
+        test_case_name = "Arithmetic Addition";
+        rppi_add_u8_pkd3_host(input, input_second, srcSize[0], output, handle);
+        break;
+    case 31:
+        test_case_name = "Arithmetic Subtraction";
+        rppi_subtract_u8_pkd3_host(input, input_second, srcSize[0], output, handle);
+        break;
+    case 32:
+        test_case_name = "Bitwise AND";
+        rppi_bitwise_AND_u8_pkd3_host(input, input_second, srcSize[0], output, handle);
+        break;
+    case 33:
+        test_case_name = "Bitwise EXCLUSIVE OR";
+        rppi_exclusive_OR_u8_pkd3_host(input, input_second, srcSize[0], output, handle);
+        break;
+    case 34:
+        test_case_name = "Bitwise INCLUSIVE OR";
+        rppi_inclusive_OR_u8_pkd3_host(input, input_second, srcSize[0], output, handle);
+        break;
+    case 35:
+        test_case_name = "Bitwise NOT";
+        rppi_bitwise_NOT_u8_pkd3_host(input, srcSize[0], output, handle);
+        break;
+    case 36:
+        test_case_name = "Box Filter";
+        rppi_box_filter_u8_pkd3_host(input, srcSize[0], output, kernelSize, handle);
+        break;
+    case 37:
+        test_case_name = "Canny Edge Detector";
```

Comment:
Codacy found an issue: [Variable 'test_case_name' is assigned a value that is never used.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/rpp/pullRequest?prid=5976402)